astropy/coordinates/builtin_frames/intermediate_rotation_transforms.py

```diff
@@ -16,11 +16,29 @@
 from .gcrs import GCRS, PrecessedGeocentric
 from .cirs import CIRS
 from .itrs import ITRS
+from .teme import TEME
 from .utils import get_polar_motion, get_jd12
 
 # # first define helper functions
 
 
+def teme_to_itrs_mat(time):
+    # Sidereal time, rotates from ITRS to mean equinox
+    gst = erfa.gmst82(*get_jd12(time, 'ut1'))
```

Comment:
I'd just use `time.sidereal_time('mean', model='IAU1982', longitude='greenwich')`

Aside: why the old sidereal time model? Is that part of the TEME definition? If so, do add a comment.

Reply:
The erfa call is significantly faster, so I'll keep that. 

The 1982 model is for consistency with the canonical implementation of TEME, which is Vallado 2006. I've added a comment.